Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -63,10 +66,13 @@ function filter(predicate)
     if (!@isObject(this))
         @throwTypeError("Iterator.prototype.filter requires that |this| be an Object.");
 
-    if (!@isCallable(predicate))
+    var iterated = this;
```

Comment:
Let's just use `this` directly in this function instead of `iterated` when possible.